src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -67,7 +67,9 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
         const VolumePtr & volume,
         const std::optional<String> & relative_path,
         Type part_type_,
-        const IMergeTreeDataPart * parent_part_);
+        const IMergeTreeDataPart * parent_part_,
+        bool adaptive = false,
```

Comment:
I don't really like this flag. At least a comment is needed.
Also, how index_granularity for projections work? It may be reasonable to have different granularity for a projection (maybe allow to specify index_granularity setting to projection later).